src/HealthChecks.UI.K8s.Operator/Handlers/DeploymentHandler.cs

```diff
@@ -0,0 +1,127 @@
+                labelSelector: $"resourceId={resource.Metadata.Uid}");
+            return deploymentList.Items.FirstOrDefault();
+        }
+
+        public async Task<V1Deployment> GetOrCreateAsync(HealthCheckResource resource)
+        {
+            var deployment = await Get(resource);
+            if (deployment != null) return deployment;
+
+            try
+            {
+                var deploymentResource = Build(resource);
+                var response =
+                    await _client.CreateNamespacedDeploymentWithHttpMessagesAsync(deploymentResource,
+                        resource.Metadata.NamespaceProperty);
+                deployment = response.Body;
+
+                _logger.LogInformation("Deployment {deployment} has been created", deployment.Metadata.Name);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Error creating deployment: {error}", ex.Message);
+            }
+
+            return deployment;
+
+        }
+
+        public async Task Delete(HealthCheckResource resource)
+        {
+            try
+            {
+                await _client.DeleteNamespacedDeploymentAsync($"{resource.Spec.Name}-deploy",
+                    resource.Metadata.NamespaceProperty);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Error deleting deployment for hc resource: {name} - err: {error}", resource.Spec.Name, ex.Message);
+            }
+        }
+
+        public V1Deployment Build(HealthCheckResource resource)
+        {
+
+            var metadata = new V1ObjectMeta
+            {
+                Labels = new Dictionary<string, string>
+                {
+                    ["resourceId"] = resource.Metadata.Uid,
+                    ["app"] = resource.Spec.Name
+                },
+                Name = $"{resource.Spec.Name}-deploy",
+                NamespaceProperty = resource.Metadata.NamespaceProperty
+            };
+
+            var spec = new V1DeploymentSpec
+            {
+                Selector = new V1LabelSelector
+                {
+                    MatchLabels = new Dictionary<string, string>
+                    {
+                        ["app"] = resource.Spec.Name
+                    }
+                },
+                Replicas = 1,
+                Template = new V1PodTemplateSpec
+                {
+                    Metadata = new V1ObjectMeta
+                    {
+                        Labels = new Dictionary<string, string>
+                        {
+                            ["app"] = resource.Spec.Name
+                        }
+                    },
+                    Spec = new V1PodSpec
+                    {
+                        Containers = new List<V1Container>
+                        {
+                            new V1Container
+                            {
+                                ImagePullPolicy = "Never",
```

Comment:
Never?

Reply:
This should be "Always", I forgot to change it when doing minikube local testing as the docker image is not published yet.